posts/2019-09-26-Rust-1.38.0.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: post
+title: "Announcing Rust 1.38.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.38.0. Rust is a programming language that is empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust 1.38.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the appropriate page on our website.
+
+[install]: https://www.rust-lang.org/install.html
+
+## What's in 1.38.0 stable
+
+### Pipelined compilation
+
+[internals-pipelined]: https://internals.rust-lang.org/t/evaluating-pipelined-rustc-compilation/10199
+
+To compile a crate, the compiler doesn't need the dependencies to be fully built. Instead, it just needs their "metadata" (i.e. the list of types, dependencies, exports...). This metadata is produced early in the compilation process. Starting with Rust 1.38.0, Cargo will take advantage of this by automatically starting to build dependent crates as soon as metadata is ready.
+
+While the change doesn't have any effect on builds for a single crate, during testing [we got reports][internals-pipelined] of 10-20% compilation speed increases for optimized, clean builds of some crate graphs. Other ones did not improve much, and the speedup depends on the hardware running the build, so your mileage might vary. No code changes are needed to benefit from this.
+
+### Linting some incorrect uses of `mem::{uninitialized, zeroed}`
+
+As [previously announced](https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html#maybeuninitt%3E-instead-of-mem::uninitialized), `std::mem::uninitialized` is hard to use soundly. Instead, `MaybeUninit<T>` should be used.
```

Comment:
The 1.36 announcement says that it is "essentially impossible" to use soundly. But it's not unsound to use to initialize plain-old-data (in the C++ sense), is it? If that's correct, then "hard to use soundly" seems reasonable; if not, then maybe we should go back to using stronger language.

Reply:
I'd be fine with "essentially impossible" as well -- I tweaked this because it sounded 100% certain but there are cases where it is OK... for example: `mem::uninitialized::<MaybeUninit<u8>>()`. ;)

Actually, let's change back to "essentially impossible".

Reply:
```suggestion
As [previously announced](https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html#maybeuninitt%3E-instead-of-mem::uninitialized), `std::mem::uninitialized` is essentially impossible to use safely. Instead, `MaybeUninit<T>` should be used.
```